src/layout-rect.js

```diff
@@ -229,3 +229,19 @@ export function layoutRectEquals(r1, r2) {
   return r1.left == r2.left && r1.top == r2.top &&
       r1.width == r2.width && r1.height == r2.height;
 }
+
+/**
+ * @param {LayoutMarginsChangeDef} marginsChange
```

Comment:
I think the type here should be {LayoutMarginsChangeDef|undefined} (same for the return type)